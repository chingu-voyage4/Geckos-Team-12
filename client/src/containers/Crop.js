import React, { Component } from "react";
import { connect } from "react-redux";
import { getCrop } from "../redux/Crop";
import Back from "../components/Back";
import "../assets/css/crop.css";

const CropHead = ({ title, image_url_header }) => (
  <div className="headbg" style={{ background: `url(${image_url_header})` }}>
    <h1 className="headText">{title} </h1>
  </div>
);

// adding description and bullet section
class CropInfo extends Component {
  render() {
    const { short_desc, difficulty_level, best_season, climate } = this.props;

    return (
      <div className="infoSection">
        <div className="infoDesc">
          <p>{short_desc}</p>{" "}
        </div>
        <div className="infoWrapper">
          <div className="infoItems">Difficulty : {difficulty_level}</div>
          <div className="infoItems">Best time to plant : {best_season}</div>
          <div className="infoItems">Suitable climate: {climate}</div>
        </div>
      </div>
    );
  }
}

// Final render of Crop page
class Crop extends Component {
  componentWillMount() {
    this.props.getCrop(this.props.match.params.name);
  }
  render() {
    const { crop, loading } = this.props;
    return (
      <div className="crop">
        <Back to="/catalog" text="Catalog" />
        {loading && <div>Loading Crop...</div>}
        {!loading && (
          <div>
            <CropHead
              title={crop.name}
              image_url_header={crop.image_url_header}
            />
            <CropInfo {...crop} />
          </div>
        )}
      </div>
    );
  }
}
const mapStateToProps = state => ({
  loading: state.crop.loading,
  crop: state.crop.crop,
  error: state.crop.error
});
const mapDispatchToProps = dispatch => ({
  getCrop: name => dispatch(getCrop(name))
});
export default connect(mapStateToProps, mapDispatchToProps)(Crop);
