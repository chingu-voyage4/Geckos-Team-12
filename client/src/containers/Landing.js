import React, { Component } from "react";
import { connect } from "react-redux";
import "../assets/css/landing.css";
import Login from "../components/Login";
import Signup from "../components/Signup";
import Toast from "../components/Toast";
import { signInUser, signUpUser } from "../redux/Auth";

export class Landing extends Component {
  constructor(props, context) {
    super(props, context);
    this.handleLogin = this.handleLogin.bind(this);
    this.handleSignup = this.handleSignup.bind(this);
  }
  handleLogin() {
    let values = this.props.form.loginForm.values;
    //console.log(values);
    this.props
      .signInUser(values)
      .then(() => this.props.history.push("/mygarden#header"))
      .catch(err => this.setState({ error: err }));
  }
  handleSignup() {
    let values = this.props.form.signupForm.values;
    //console.log(values);
    this.props
      .signUpUser(values)
      .then(() => this.props.history.push("/mygarden#header"))
      .catch(err => this.setState({ error: err }));
  }
  render() {
    const { showToast, toastMessage, toastError } = this.props;
    return (
      <div>
        <div className="landing">
          <div className="s1">
            <div className="catchy-phrase">
              <h1>Grow anything simply.</h1>
              <h2>SimpleSeed helps you easily grow your own organic food.</h2>
            </div>
            <div className="catchy-footer">
              <div className="catchy-feature">
                <div className="catchy-img">
                  <i className="fab fa-leanpub fa-5x" />
                </div>
                <div className="catchy-content">
                  <p>Learn growing your own food.</p>
                </div>
              </div>
              <div className="catchy-feature">
                <div className="catchy-img">
                  <i className="far fa-bell fa-5x" />
                </div>
                <div className="catchy-content">
                  <p>Get specific reminders.</p>
                </div>
              </div>
              <div className="catchy-feature">
                <div className="catchy-img">
                  <i className="fas fa-trophy fa-5x" />
                </div>
                <div className="catchy-content">
                  <p>Earn community badges</p>
                </div>
              </div>
            </div>
          </div>

          <div className="s2">
            <h1>How does it work?</h1>
            <iframe
              title="SimpleSeed HowTo"
              width="560"
              height="315"
              src="https://www.youtube.com/embed/9IB1V9H0K6s"
            />
          </div>
          <div className="s3">
            <h1 className="features-title">Why use SimpleSeed?</h1>
            <div className="feature">
              <div className="feature-img">
                <i className="fab fa-leanpub fa-5x" />
              </div>
              <div className="feature-content">
                <p>
                  Browse freely our collection of crop basic info and tutorials.
                  Our dedicated coders-gardeners have gathered data from across
                  th web to help you learn how to easily grow organic food.
                </p>
              </div>
            </div>
            <div className="feature-flip">
              <div className="feature-img-flip">
                <i className="far fa-bell fa-5x" />
              </div>
              <div className="feature-content-flip">
                <p>
                  Busy with work and family. No worries, we got you covered we
                  will remind everytime your crops need attention. You can
                  choose from web to email notifications.
                </p>
              </div>
            </div>
            <div className="feature">
              <div className="feature-img">
                <i className="fas fa-trophy fa-5x" />
              </div>
              <div className="feature-content">
                <p>
                  Complete our easy tutorials and earn badges and trophys to
                  become a an active member of our community. And if you want it
                  you can start writing your own tutorials.
                </p>
              </div>
            </div>
            <div className="feature-flip">
              <div className="feature-img-flip">
                <i className="fas fa-users fa-5x" />
              </div>
              <div className="feature-content-flip">
                <p>
                  Be part of a growing community. More and more people Turn to
                  their yard to grow their own crops. From aromatic herbds to
                  potatoes we are here to help you join ypur neighbours into
                  this great experience.
                </p>
              </div>
            </div>
          </div>

          <div className="s4" id="join">
            <h1>Give it a try</h1>
            <div className="s4-1">
              <div className="form-title">Login</div>
              <Login
                handleSubmit={e => {
                  e.preventDefault();
                  this.handleLogin();
                }}
              />
            </div>
            <div className="divider" />
            <div className="s4-2">
              <div className="form-title">SignUp</div>
              <Signup
                handleSubmit={e => {
                  e.preventDefault();
                  this.handleSignup();
                }}
              />
            </div>
          </div>
        </div>
        <Toast show={showToast} message={toastMessage} error={toastError} />
      </div>
    );
  }
}
const mapDispatchToProps = dispatch => ({
  signInUser: values => dispatch(signInUser(values)),
  signUpUser: values => dispatch(signUpUser(values))
});
const mapStateToProps = state => {
  return {
    auth: state.auth.authenticated,
    showToast: state.auth.showToast,
    toastMessage: state.auth.toastMessage,
    toastError: state.auth.toastError,
    form: state.form
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Landing);
