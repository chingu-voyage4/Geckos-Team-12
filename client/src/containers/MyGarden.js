import React, { Component } from "react";
import "../assets/css/mygarden.css";
class MyGarden extends Component {
  componentWillMount() {
    window.scrollTo(0, 0);
  }
  render() {
    return (
      <div className="mygarden">
        <div className="mygarden-header" id="header">
          <div className="mygarden-header-left">
            My<span>Garden</span>
          </div>
          <div className="mygarden-header-right">
            <h2>Welcome to your garden</h2>
            <h4>Here are some stats</h4>
            <ul>
              <li>
                You have completed <span>3 tutorials</span> since you joigned.
              </li>
              <li>
                You have planted <span>12 crops.</span>
              </li>
              <li>
                You have harvested <span>2 differents crops </span> so far this
                year.
              </li>
            </ul>
          </div>
        </div>
        <div className="mygarden-body">
          <div className="mygarden-body-top">
            <div className="mygarden-body-user">User</div>
            <div className="mygarden-body-today-tasks">Today's Tasks</div>
          </div>
          <div className="mygarden-body-center">
            <div className="mygarden-body-week-tasks">Week's Tasks</div>
          </div>
          <div className="mygarden-body-bottom">
            <div className="mygarden-body-crops">MyCrops</div>
            <div className="mygarden-body-suggestion">Crop Suggestions</div>
          </div>
        </div>
      </div>
    );
  }
}

export default MyGarden;
