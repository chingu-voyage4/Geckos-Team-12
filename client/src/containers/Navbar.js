import React, { Component } from "react";
import { Link, withRouter } from "react-router-dom";
import { connect } from "react-redux";

import { signInUser, signOutUser } from "../redux/Auth";
import "../assets/css/navBar.css";

export class Navbar extends Component {
  signOut = () => this.props.signOut;
  handleSignIn = () => {
    this.props.signIn();
    this.props.history.push("/mygarden");
  };
  handleSignOut = () => {
    this.props.signOut();
    this.props.history.push("/");
  };
  render() {
    const { auth } = this.props;
    return (
      <div className="nav-bar">
        <div
          className="nav-bar-brand"
          onClick={() => this.props.history.push("/")}
        >
          SimpleSeed
        </div>
        <div className="nav-bar-menu">
          <Link className="" to="/catalog">
            Catalog
          </Link>
          <Link className="" to="/mygarden">
            MyGarden
          </Link>
        </div>
        <div className="nav-bar-user">
          {!auth && (
            <div className="signin">
              <button onClick={this.handleSignIn}>Sign In</button>
              {/* <Link to="/signin">SignIn</Link> */}
              <Link to="/signup">SignUp</Link>
            </div>
          )}
          {auth && (
            <div className="user">
              <button className="signout" onClick={this.handleSignOut}>
                Sign Out
              </button>
              <button
                className="profile"
                to="/profile"
                onClick={() => this.props.history.push("/profile")}
              />
            </div>
          )}
        </div>
      </div>
    );
  }
}
const mapStateToProps = state => {
  return {
    auth: state.auth.authenticated
  };
};
const mapDispatchToProps = dispatch => ({
  signIn: () => dispatch(signInUser()),
  signUp: () => dispatch(signInUser()),
  signOut: () => dispatch(signOutUser())
});

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(Navbar));
