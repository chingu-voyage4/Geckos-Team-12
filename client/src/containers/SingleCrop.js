import React, { Component } from "react";
import { withRouter } from "react-router-dom";

let heroImg = "https://images.unsplash.com/photo-1467020323552-36f7bf0e30e6?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=531319c82648653fd61091ddd2141768&auto=format&fit=crop&w=750&q=80";

class CropHead extends Component{
	constructor(props){
		super(props);
		this.state = {
      headerText: "Tomato",
      headerImg: heroImg;
		}
    render(){
      return (
        <CropHead />  
      )
    }
	}
}


// Final render of Crop page
class SingleCrop extends Component {
	<CropHead/>		
}

export default SingleCrop;
