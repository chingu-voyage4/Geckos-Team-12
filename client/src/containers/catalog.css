@import url('https://fonts.googleapis.com/css?family=Montserrat|Pacifico');

*{
    margin: 0;
    padding:0;
    box-sizing:borderbox;
  }
  html{
    font-size:62.5%;
    font-family: 'Montserrat', sans-serif;
  }
  
  body{
      font-size: 1.5rem;
      color: white;
    height:100vh;
    width: 100vw;

  
    background-image: linear-gradient(to right bottom, rgba(126, 213, 111, 0.5), rgba(40, 180, 133, 0.8)),url(https://images.unsplash.com/photo-1457530378978-8bac673b8062?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=a13230dddd3ff4b3150eab5706e8c9ae&auto=format&fit=crop&w=1050&q=80);
    background-size:cover;
    background-repeat:no-repeat;
    
  }
  nav{
    display:flex;
    justify-content:space-between;
  }

  nav a, h3{
   
        margin:1rem 4rem;
        text-decoration: none;
        color: #fff;
  }

  nav .logo{
      font-family: 'Pacifico', cursive;
      font-size:2.5rem;
}

  .choose{
 
    display: flex;
    align-items: center;
    margin:3rem 0;
  
    flex-direction: column;
}

img{
    width:10rem;
    height:10rem;
    /* background-size:cover; */

}

.crop-container{
display: flex;
text-align: center;
justify-content: center;
/* display: none; */
margin: 3rem;

}

input{
    background: rgba(259, 259, 259, .5);
    height:2.5rem;
    outline: none;
    border: 0;
    width: 35%;
    margin: 1rem;
    } 