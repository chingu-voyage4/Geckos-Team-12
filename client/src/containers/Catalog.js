import React, { Component } from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import { getCrops } from "../redux/Crops";
import "../assets/css/catalog.css";

export class Catalog extends Component {
  componentWillMount() {
    this.props.getCrops();
  }
  render() {
    const { crops, loading } = this.props;
    return (
      <div className="container">
        <section className="choose">
          <h1>Choose your crop</h1>
          <input type="text" />
        </section>
        <div className="crop-container">
          {loading && <p>Crops Loading...</p>}
          {!loading && crops.length > 0
            ? crops.map(crop => (
                <div key={crop._id}>
                  <div className="crop-data">
                    <Link to={`/crops/${crop._id}`}>
                      <img src={crop.image_url_thumb} alt="Crop thumb" />
                      <h3>{crop.name}</h3>
                      <p>{crop.category}</p>
                    </Link>
                  </div>
                </div>
              ))
            : !loading && (
                <p className="crop-error">
                  No crops found...Come back next spring
                </p>
              )}
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    crops: state.crops.crops,
    loading: state.crops.loading,
    auth: state.auth.authenticated
  };
};

const mapDispatchToProps = dispatch => ({
  getCrops: () => dispatch(getCrops())
});

// function changeImage(){
//  let
// }
export default connect(mapStateToProps, mapDispatchToProps)(Catalog);
