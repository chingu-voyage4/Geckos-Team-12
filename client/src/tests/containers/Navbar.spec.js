/* global shallow, toJson */
import React from "react";
import { Navbar } from "../../containers/Navbar";

const setup = props => {
  const wrappedComponent = shallow(<Navbar {...props} />);

  return {
    props,
    wrappedComponent
  };
};

describe("Navbar", () => {
  it("should render", () => {
    const props = {};
    const { wrappedComponent } = setup(props);
    expect(toJson(wrappedComponent)).toMatchSnapshot();
  });
});
