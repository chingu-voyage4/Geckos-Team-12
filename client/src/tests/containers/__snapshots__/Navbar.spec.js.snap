// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render 1`] = `
<div
  className="nav-bar"
>
  <div
    className="nav-bar-brand"
    onClick={[Function]}
  >
    SimpleSeed
  </div>
  <div
    className="nav-bar-menu"
  >
    <Link
      className=""
      replace={false}
      to="/catalog"
    >
      Catalog
    </Link>
    <Link
      className=""
      replace={false}
      to="/mygarden"
    >
      MyGarden
    </Link>
  </div>
  <div
    className="nav-bar-user"
  >
    <div
      className="signin"
    >
      <button
        onClick={[Function]}
      >
        Sign In
      </button>
      <Link
        replace={false}
        to="/signup"
      >
        SignUp
      </Link>
    </div>
  </div>
</div>
`;
